core/txpool/legacypool/list.go

```diff
@@ -342,9 +342,12 @@ func (l *list) Add(tx *types.Transaction, priceBump uint64, l1CostFn txpool.L1Co
 		return false, nil
 	}
 	l.totalcost.Add(l.totalcost, cost)
-	if l1CostFn != nil {
-		if l1Cost := l1CostFn(tx.RollupCostData()); l1Cost != nil { // add rollup cost
-			l.totalcost.Add(l.totalcost, cost)
```

Comment:
I believe we had a longstanding bug in here, where we would just never add the actual `l1Cost` but the execution cost a second time.

Reply:
I believe this was fixed in https://github.com/ethereum-optimism/op-geth/pull/324 but we never merged it

Reply:
Wow it was indeed, and we totally dropped the ball on it.

Reply:
Closed that PR and gave the author credit.